### PR TITLE
DOC: add build instructions

### PR DESCRIPTION
**Pull request by pv**
Move Scipy build instructions from the scipy.org web page to the documentation here.

Also scrap 6+ year old Linux instructions that are likely to be more confusing than helpful today.

CircleCI: https://4803-1460385-gh.circle-artifacts.com/0/home/ubuntu/scipy/doc/build/html-scipyorg/index.html

### CODE REVIEW

File: doc/source/building/windows.rst [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+--------
+
+Compared to OSX and Linux, building NumPy and SciPy on Windows is more difficult,
+largely due to the lack of compatible, open-source libraries like LAPACK_ or
+ATLAS_ that are necessary to build both libraries and have them perform
+relatively well. It is not possible to just call a one-liner on the command
+prompt as you would on other platforms via ``sudo apt-get install`` machinery.
+
+Fortunately, a lot of work has been done recently to rectify this situation.
+Projects such as OpenBLAS_ and Mingwpy_ are under active development to develop
```
Comment (xoviat) [inline doc/source/building/windows.rst]: Mingwpy has been abandoned.

File: doc/source/building/linux.rst [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+    sudo dnf install gcc-gfortran python3-devel python2-devel openblas-devel lapack-devel Cython
+
+
+Intel C compiler and MKL
+========================
+
+Intel MKL 11.0 (updated Dec 2012)
+---------------------------------
+
+Add the following lines to site.cfg in your top level NumPy directory to use Intel® MKL for Intel® 64 (or earlier known as em64t) architecture, considering the default installation path of Intel® MKL which is bundled with Intel® Composer XE SP1 version on Linux:
```
Comment (person142) [inline doc/source/building/linux.rst]: Nitpick: break up the long lines?

File: doc/source/building/linux.rst [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+or
+
+::
+
+   python setup.py build
+   python setup.py install --prefix=$HOME/local
+
+Before building, you will also need to install packages that NumPy and SciPy depend on
+
+* BLAS and LAPACK libraries (optional but strongly recommended for NumPy, required for SciPy): typically `ATLAS <http://math-atlas.sourceforge.net/>`__ + `LAPACK <http://www.netlib.org/lapack/>`__, or `MKL <https://software.intel.com/en-us/intel-mkl>`__, or `ACML <http://developer.amd.com/tools-and-sdks/archive/amd-core-math-library-acml/>`__
```
Comment (rgommers) [inline doc/source/building/linux.rst]: Remove ACML and LAPACK, add OpenBLAS?

File: doc/source/building/linux.rst [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+
+See /usr/share/doc/libatlas3-base/README.Debian for instructions on how to
+build optimized ATLAS packages for your specific CPU.
+The packaged OpenBLAS chooses the optimal code at runtime so it does not need
+recompiling unless the packaged version does not yet support the used CPU.
+
+As of Jan. 2014 ATLAS is the recommended library to use as OpenBLAS will
+deadlock when used in combination with the `multiprocessing` module
+(or any use of `os.fork`) and older versions (<= 0.2.8) tend to crash when used
+on larger problem sizes.
```
Comment (rgommers) [inline doc/source/building/linux.rst]: paragraph no longer relevant

File: doc/source/building/linux.rst [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+
+   $export LD_LIBRARY_PATH=/opt/intel/composer_xe_2013/mkl/lib/intel64: /opt/intel/composer_xe_2013/compiler/lib/intel64:$LD_LIBRARY_PATH
+
+If you build NumPy for ia32 bit platforms:
+
+::
+
+   $export LD_LIBRARY_PATH=/opt/intel/composer_xe_2013/mkl/lib/ia32: /opt/intel/composer_xe_2013/compiler/lib/ia32:$LD_LIBRARY_PATH
+
+Intel MKL 10.0
```
Comment (rgommers) [inline doc/source/building/linux.rst]: too old, section can be deleted

File: doc/source/building/macosx.rst [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+Though virtually any commercial C/C++ compiler may be used with SciPy, OS X
+comes with GNU C compilers pre-installed. The only thing missing is the GNU
+FORTRAN compiler.
+
+We recommend gfortran; this is a free, open source, F95 compiler. We suggest you
+use the following binaries:
+
+* For Snow Leopard : https://cran.r-project.org/bin/macosx/tools/gfortran-4.2.3.pkg
+* For Lion : http://r.research.att.com/gfortran-lion-5666-3.pkg (for
+  Xcode 4.1)
```
Comment (rgommers) [inline doc/source/building/macosx.rst]: Snow Leopard & Lion can be deleted.

File: doc/source/building/macosx.rst [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+* Later versions : http://r.research.att.com/tools/gcc-42-5666.3-darwin11.pkg (for Xcode
+  4.2 or higher) (also available through Homebrew)
```
Comment (rgommers) [inline doc/source/building/macosx.rst]: I think the better recommendation is Homebrew now, has gfortran 4.9 that works well.

File: doc/source/building/macosx.rst [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+<http://cython.org/>`__ compiler is also needed.
+
+Version-specific notes
+----------------------
+
+This section notes only things specific to one version of OS X or Python. 
+The build instructions in :ref:`Obtaining and Building NumPy and SciPy
+<osx-obtaining-and-building>` apply to all versions.
+
+OS X 10.7 (Lion) and 10.8 (Mountain Lion)
```
Comment (rgommers) [inline doc/source/building/macosx.rst]: Whole section can be removed.

File: doc/source/building/macosx.rst [hunk 9]
```diff
@@ -0,0 +1,188 @@
...
+::
+
+     $ export CC=clang
+     $ export CXX=clang++
+     $ export FFLAGS=-ff2c
+
+Alternatively, you may try installing gcc-4.2 manually, and then using
+the environment variables in the prior block.
+
+Python 2.6
```
Comment (rgommers) [inline doc/source/building/macosx.rst]: can be removed as well

File: doc/source/building/macosx.rst [hunk 10]
```diff
@@ -0,0 +1,188 @@
...
+
+  * ``$ gcc --version``
+
+  * ``$ gfortran --version``
+
+* The versions of numpy and scipy that you are trying to install
+
+* The full output of ``$ python setup.py build``
+
+Older notes
```
Comment (rgommers) [inline doc/source/building/macosx.rst]: section can be removed

**Comment by rgommers** [comment]
Good idea, like the new Windows instructions!

**Comment by pv** [comment]
Scrapped more obsolete stuff.

**Pull request merged**